grammar/parsing.js

```diff
@@ -206,12 +206,47 @@ class Parsing extends EmbeddedActionsParser {
             return result;
         });
 
+        $.RULE('multiArray', () => {
+            var multiArr = null;
+            let singleArr = null
+            $.CONSUME(OpenCurlyParen);
+            let single = $.OR([
+                {ALT: () => {
+                    multiArr = []
+                    multiArr.push($.SUBRULE($.subArray))
+                    return false
+                }},
+                {ALT: () => {
+                    singleArr = $.SUBRULE($.constantArray);
+                    $.CONSUME(CloseCurlyParen)
+                    return true;
+                }},
+            ])
+            if(single === true){
```

Comment:
`if(single)`

Reply:
when assessing or statements, chevrotain runs "simulations" on the code to pick the best path (its black boxed). when it does this, the $.OR statement returns a string, rather than either of my Booleans and thus I need to check if it === true. This leads to situations where single is a string, even though the "path" chevrotain took is the false path.

TLDR: chevrotain does a weird thing where it returns a non Boolean to single and thus I need to === true b/c sometimes its a string.

Reply:
Oof that's nasty, thanks for the comment!

Reply:
Very nice code comment. Couple nits on the whitespace (this is consistent throughout the app): 

```suggestion
            if (single === true) {
```

Similarly, spaces after the comments: 
```
// chevrotain occasionally ...
```